src/data/ACTIONS/root/SGE.ts

```diff
@@ -1,5 +1,298 @@
+import {Attribute} from 'event'
 import {ensureActions} from '../type'
 
+const SGE_STUB = 190705000
+
 export const SGE = ensureActions({
-	// TODO: Add actions here
+	/** Single-Target Heals/Shields */
+	DIAGNOSIS: {
+		id: SGE_STUB + 1,
+		name: 'Diagnosis',
+		icon: '',
+		onGcd: true,
+		speedAttribute: Attribute.SPELL_SPEED,
+		castTime: 1500,
+		mpCost: 400,
+	},
+	EUKRASIAN_DIAGNOSIS: {
+		id: SGE_STUB + 2,
+		name: 'Eukrasian Diagnosis',
+		icon: '',
+		onGcd: true,
+		gcdRecast: 1500,
+		mpCost: 1000,
+		statusesApplied: ['EUKRASIAN_DIAGNOSIS', 'DIFFERENTIAL_DIAGNOSIS'],
+	},
+	HAIMA: {
+		id: SGE_STUB + 3,
+		name: 'Haima',
+		icon: '',
+		cooldown: 120000,
+		statusesApplied: ['HAIMATINON'],
+	},
+	EGEIRO: {
+		id: SGE_STUB + 4,
+		name: 'Egeiro',
+		icon: '',
+		onGcd: true,
+		speedAttribute: Attribute.SPELL_SPEED,
+		castTime: 8000,
+		mpCost: 2400,
+	},
+
+	/** AoE Heals/Shield */
+	PROGNOSIS: {
+		id: SGE_STUB + 5,
+		name: 'Prognosis',
+		icon: '',
+		onGcd: true,
+		speedAttribute: Attribute.SPELL_SPEED,
+		castTime: 2000,
+		mpCost: 1000,
+	},
+	EUKRASIAN_PROGNOSIS: {
+		id: SGE_STUB + 6,
+		name: 'Eukrasian Prognosis',
+		icon: '',
+		onGcd: true,
+		gcdRecast: 1500,
+		mpCost: 1100,
+		statusesApplied: ['EUKRASIAN_PROGNOSIS'],
+	},
+	PANHAIMA: {
+		id: SGE_STUB + 7,
+		name: 'Panhaima',
+		icon: '',
+		cooldown: 120000,
+		statusesApplied: ['PANHAIMATINON'],
+	},
+	PHYSIS: {
+		id: SGE_STUB + 8,
+		name: 'Physis',
+		icon: '',
+		cooldown: 60000,
+		statusesApplied: ['PHYSIS'],
+	},
+	PHYSIS_II: {
+		id: SGE_STUB + 9,
+		name: 'Physis II',
+		icon: '',
+		cooldown: 180000,
+		statusesApplied: ['PHYSIS_II'],
+	},
+	HOLOS: {
+		id: SGE_STUB + 10,
+		name: 'Holos',
+		icon: '',
+		cooldown: 60000,
+	},
+	PEPSIS: {
+		id: SGE_STUB + 11,
+		name: 'Pepsis',
+		icon: '',
+		cooldown: 15000,
+	},
+
+	/** Utility */
+	KARDIA: {
+		id: SGE_STUB + 12,
+		name: 'Kardia',
+		icon: '',
+		cooldown: 5000,
+	},
+	EUKRASIA: { // TODO: Apparently this is on the GCD? Will need confirmation after launch
```

Comment:
Looks like, based on its classification as a Spell. If I had to guess, I'd say this will probably work a lot like the Mudra system - e.g. Eurkasia (1s) > Diagnosis (1.5s) for a 2.5s cast, much like Ten (0.5s) > Chi (0.5s) > Raiton (1.5s) is a 2.5s cast. Also my gut says this will probably apply a buff that Dosis/Diagnosis/Prognosis consumes but we'll see when EW launches I guess.

Reply:
Yeah, on re-watching footage/re-checking Balance notes, it's applying a buff, and it's a Eukrasia (Instant, 1s GCD) > [Eukrasia Consumer] (Instant, 1.5s GCD) for a total of a 2.5s GCD (notably unaffected by Spell Speed though). Added the status and tagged it in Eukrasia's `statusesApplied`